Assets/Prefabs/Game Scripts/Player.cs

```diff
@@ -14,19 +18,29 @@ void Start()
         _rigidbody = GetComponent<Rigidbody>();
     }
 
-    // Update is called once per frame
-    void Update()
+    public void Update()
     {
         
+        // If we are in range, reset the action
+        if (targetPosition != null && Vector3.Distance((Vector3) targetPosition, transform.position) < range)
```

Comment:
nit, but this seems like something that should be done on fixedupdate instead of every frame

Reply:
Very true, fixed